mlir/include/mlir/IR/Matchers.h

```diff
@@ -100,6 +101,41 @@ struct constant_op_binder {
   }
 };
 
+/// A matcher that matches operations that implement the
+/// `InferIntRangeInterface` interface, and binds the inferred range.
+struct infer_int_range_op_binder {
+  IntegerValueRange *bind_value;
+
+  infer_int_range_op_binder(IntegerValueRange *bind_value)
```

Comment:
```suggestion
  explicit infer_int_range_op_binder(IntegerValueRange *bind_value)
```
or should this really be implicitly constructible?

Reply:
Good point!